### DISCUSSION TITLE
props and layerName are not found in Composite layer

### DISCUSSION THREAD

**Discussion by neelduttahere** [visgl/deck.gl#6564] [Q&A]
Hi, 
I am just learning how to implement composite layers. Below is the code I am trying to run:

```js
import {CompositeLayer, TextLayer} from "@deck.gl/layers";

class WrappedTextLayer extends CompositeLayer {
    renderLayers() { // a method of `Layer` classes
        // special logic here
        return [new TextLayer(this.props)];
    }
}
WrappedTextLayer.layerName = 'WrappedTextLayer';
WrappedTextLayer.defaultProps = {
    getText: (): string => 'x',
    getSize: (): number => 32,
    getColor: [255, 255, 255]
};

export { WrappedTextLayer };
```

However, props are showing as error, with the message `Property 'props' does not exist on type 'WrappedTextLayer'.ts(2339)`.
Same goes for .layerName and .defaultProps, with their error messages respectively:
1. `Property 'layerName' does not exist on type 'typeof WrappedTextLayer'.ts(2339)`
2. `Property 'defaultProps' does not exist on type 'typeof WrappedTextLayer'.ts(2339)`

What am I doing wrong? Have I imported incorrectly?

FYI: The above code snippet is from https://stackoverflow.com/questions/68532651/react-deck-gl-add-default-props-to-each-child-component/70755312#70755312 which I am trying to learn and implement from.

**Comment by Pessimistress** [comment]
https://github.com/danmarshall/deckgl-typings
